### PR TITLE
Allow submitqueue to merge on either jenkins ci or shippable green

### PR DESCRIPTION
**Pull request by jlowdermilk**
cc @brendandburns, @lavalamp

### CODE REVIEW

File: mungegithub/pulls/submit-queue.go [hunk 1]
```diff
@@ -460,16 +474,18 @@ func (sq *SubmitQueue) doGithubE2EAndMerge(pr *github_api.PullRequest) {
 
 	// Wait for the build to start
 	sq.SetPRStatus(pr, ghE2EWaitingStart)
-	err = sq.githubConfig.WaitForPending(pr)
+	err = sq.githubConfig.WaitForPending(pr, []string{gceE2EContext})
```
Comment (eparis) [inline mungegithub/pulls/submit-queue.go]: Shouldn't this also be `sq.E2EStatusContext`  (yes, they are the same)

**Comment by eparis** [comment]
Should this logic also move somehow over to ping_ci as well?  Do we need to close/reopen if shippable missed running but we have jenkins ci integration running?

Does the jenkins CI run everything shippable does?

File: mungegithub/pulls/submit-queue.go (hunk 1 above)
Comment (jlowdermilk) [inline mungegithub/pulls/submit-queue.go]: Done.

**Comment by jlowdermilk** [comment]
Yes, jenkins ci runs everything shippable does. I updated ping_ci to skip if the pr has jenkins ci results.

**Comment by wojtek-t** [comment]
Having this PR would be really helpful!
Thanks a lot for working on it @jlowdermilk !

File: mungegithub/github/github.go [hunk 2]
```diff
@@ -479,9 +479,9 @@ func timeout(sleepTime time.Duration, c chan bool) {
 	c <- true
 }
 
-func (config *Config) doWaitStatus(pr *github.PullRequest, pending bool, c chan error) {
+func (config *Config) doWaitStatus(pr *github.PullRequest, pending bool, requiredContexts []string, c chan error) {
```
Comment (lavalamp) [inline mungegithub/github/github.go]: Could the contexts be added to the config to avoid passing the parameter everywhere?

Comment (lavalamp) [inline mungegithub/github/github.go]: I see the answer is no.

**Comment by eparis** [comment]
@lavalamp unless you see a problem, I'm going to merge and kick the bot off again....

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
@eparis I take it someone gave you the keys? :)

**Comment by eparis** [comment]
no comment.

**Pull request merged**